torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -416,5 +421,85 @@ GraphExecutorState ProfilingGraphExecutorImpl::getDebugState() {
   return state;
 }
 
+Node* createFallbackGraph(Block* b, ArrayRef<Value*> inputs, Graph* g) {
+  auto graph = std::make_shared<Graph>();
+  auto value_map = [](Value* v) { return v; };
+  graph->block()->cloneFrom(b, value_map);
+
+  auto fallback = g->create(prim::FallbackGraph, inputs, b->outputs().size());
+  fallback->g_(attr::Subgraph, graph);
+
+  for (size_t i = 0; i < b->outputs().size(); i++) {
+    fallback->output(i)->setType(b->outputs()[i]->type());
+    fallback->output(i)->copyMetadata(b->outputs()[i]);
+  }
+  return fallback;
+}
+
+static Function* createFallbackPathFunction(
+    Block* b,
+    const std::string& function_name) {
+  auto value_map = [](Value* v) { return v; };
+  auto graph = std::make_shared<Graph>();
+  graph->block()->cloneFrom(b, value_map);
+
+  auto otypes = c10::fmap(
+      graph->return_node()->inputs(), [](Value* v) { return v->type(); });
+  auto tuple_type = TupleType::create(otypes);
+  auto return_tuple = graph->createTuple(graph->return_node()->inputs());
+  graph->appendNode(return_tuple);
+  for (int i = graph->outputs().size() - 1; i >= 0; i--) {
+    graph->eraseOutput(i);
+  }
+  graph->registerOutput(return_tuple->output());
+  return new GraphFunction(function_name, graph, nullptr);
+}
+
+Node* insertFallbackFunctionCall(
+    Graph* graph,
+    Function* func,
+    ArrayRef<Value*> inputs) {
+  auto tuple_type = func->graph()->return_node()->input(0)->type();
+  Value* fn_constant = graph->insertNode(graph->create(prim::Constant))
+                           ->s_(attr::name, func->name())
+                           ->i_(Symbol::attr("fallback"), 1)
+                           ->output()
+                           ->setType(FunctionType::create(func));
+  std::vector<Value*> func_call_inputs = {fn_constant};
+  func_call_inputs.insert(func_call_inputs.end(), inputs.begin(), inputs.end());
+  Value* result =
+      graph->insertNode(graph->create(prim::CallFunction, func_call_inputs))
+          ->output()
+          ->setType(tuple_type);
+
+  auto fun_unpack_tuple = graph->insertNode(graph->createTupleUnpack(result));
+  return fun_unpack_tuple;
+}
+
+void ProfilingGraphExecutorImpl::replaceFallbackGraphWithFallbackFunction(
+    Block* b) {
+  Stack s;
+  for (auto it = b->nodes().begin(); it != b->nodes().end();) {
+    if (it->kind() == prim::FallbackGraph) {
+      auto fallback_func = createFallbackPathFunction(
+          it->g(attr::Subgraph)->block(), "fallback_function");
+      fallback_func->get_executor().getPlanFor(s, bailout_depth - 1);
+      fallback_functions_.emplace_back(fallback_func);
```

Comment:
fallback_func_ are responsible for owning this memory, correct ?

Reply:
yes, it's very similar to how Interpreter does it and seems pretty straightforward.